circus/plugins/watchdog.py

```diff
@@ -182,7 +183,12 @@ def _decode_received_udp_message(self, data):
         :return: decoded message
         :rtype: dict or None
         """
-        result = re.match(self.msg_regex, data)
+        """regex is created with string,
+        but bytes are received from the socket,
+        that cause exception.
+        Probably this can be done better,
+        that just decode data"""
+        result = re.match(self.msg_regex, data.decode())
```

Comment:
It makes perfect sense, I think you can even remove your comment (the method is already named `decode_received_udp_message` ;) .